invisible_cities/reco/corrections_new.py

```diff
@@ -200,10 +184,111 @@ def lt_xy_corrections(X : np.array, Y : np.array, Z : np.array, maps : ASectorMa
     LT  = get_maps_coefficient(X,Y)
     return correct_lifetime_(Z,LT)
 
-def apply_all_correction_single_maps(map_e0     : ASectorMap,
-                                     map_lt     : ASectorMap,
-                                     map_te     : Optional[ASectorMap] = None,
-                                     apply_temp : bool                 = True) -> Callable:
+def get_df_to_z_converter(map_te: ASectorMap) -> Callable:
+    """
+    For given map, it returns a function that provides the conversion
+    from drift time to z position using the mean drift velocity.
+
+    Parameters
+    ----------
+    map_te : AsectorMap
+        Correction map with time evolution of some kdst parameters.
+
+    Returns
+    -------
+    A function that returns z converted array for a given drift time input
+    array.
+    """
+
+    dv_vs_time = map_te.t_evol.dv
+    dv         = np.mean(dv_vs_time)
+    def df_to_z_converter(dt):
+        return dt*dv
+
+    return df_to_z_converter
+
+
+class norm_strategy(AutoNameEnumBase):
+    mean   = auto()
+    max    = auto()
+    custom = auto()
+
+def get_normalization_factor(map_e0    : ASectorMap,
+                             norm_strat: norm_strategy   = norm_strategy.max,
+                             norm_value: Optional[float] = None
+                             ) -> float:
+    """
+    For given map, it returns a factor that provides the conversion
+    from pes time to kr energy scale using the selected normalization
+    strategy.
+
+    Parameters
+    ----------
+    map_e0 : AsectorMap
+        Correction map for geometric corrections.
+    norm_strat : norm_strategy
+        Normalization strategy used when correcting the energy.
+    norm_value : float (Optional)
+        Normalization scale when custom strategy is selected.
+
+    Returns
+    -------
+    Factor for the kr energy scale conversion.
+    """
+    if norm_strat is norm_strategy.max:
+        norm_value =  map_e0.e0.max().max()
+    elif norm_strat is norm_strategy.mean:
+        norm_value = np.mean(np.mean(map_e0.e0))
+    elif norm_strat is norm_strategy.custom:
+        if norm_value is None:
+            s  = "If custom strategy is selected for normalization"
+            s += " user must specify the norm_value"
+            raise ValueError(s)
+    else:
+        s  = "None of the current available normalization"
+        s += " strategies was selected"
+        raise ValueError(s)
+
+    return norm_value
+
+def get_pes_to_krscale_factor(map_e0    : ASectorMap,
+                              norm_strat: norm_strategy   = norm_strategy.max,
```

Comment:
I think kripton scale normalization should just be one more option in norm_strategy, no need for a separate function